test/e2e/e2eutil.go

```diff
@@ -28,8 +29,15 @@ func RegisterTypesWithFramework(newTypes ...runtime.Object) error {
 	return nil
 }
 
-func CreateRuntimeObject(obj runtime.Object, ctx *f.TestCtx) error {
-	return f.Global.Client.Create(context.TODO(), obj, &f.CleanupOptions{TestContext: ctx})
+func CreateOrUpdateMongoDB(mdb *mdbv1.MongoDB, ctx *f.TestCtx) error {
```

Comment:
I think we discussed this at some point.

The proposed process is:

a) creation
1. Check if the resource exists
2. If is not found create

b) update
1. Check if the process exists
2. If found, Update

There's always 2 requests in this case. What if we do:

1. Create
2. If error with conflict, Update instead

With this strategy, create is always 1 request, while Update is always 2.

Reply:
@rodrigovalin the issue with this approach is we run into "resource version does not match" during update, as we first need to GET the resource, which populates the resourceVersion field. (empty string is invalid for the resource)

Reply:
Ok, then you need to consider that case because from what I see, after calling `Get` you are not using that value (`&mdbv1.MongoDB{}`) that is forgotten right after the function call. This is, it is still the "original" unmodified `mdb` object the one you are passing to the API.

Reply:
I addressed this issue in PR: #27

I didn't realise it was a problem until I hit it there.